network/tlskc.go

```diff
@@ -0,0 +1,99 @@
+package network
+
+import (
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/tls"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"errors"
+	"math/big"
+	"time"
+
+	"net"
+
+	"github.com/dedis/cothority/log"
+)
+
+type TLSCertPEM []byte
+type TLSKeyPEM []byte
+
+// NewTLSCert returns a x509-certificate valid for all CommonNames.
+func NewTLSCert(serial *big.Int, country, org, orgUnit string,
+	validYear int, subjectKeyID []byte,
+	ips []net.IP) *x509.Certificate {
+	return &x509.Certificate{
+		SerialNumber: serial,
+		Subject: pkix.Name{
+			Country:            []string{country},
+			Organization:       []string{org},
+			OrganizationalUnit: []string{orgUnit},
+			CommonName:         "*",
+		},
+		NotBefore:             time.Now(),
+		NotAfter:              time.Now().AddDate(validYear, 0, 0),
+		SubjectKeyId:          subjectKeyID,
+		BasicConstraintsValid: true,
+		IsCA:        true,
+		IPAddresses: ips,
+		ExtKeyUsage: []x509.ExtKeyUsage{x509.ExtKeyUsageClientAuth, x509.ExtKeyUsageServerAuth},
+		KeyUsage: x509.KeyUsageDigitalSignature | x509.KeyUsageCertSign |
```

Comment:
Mhhh, since you coded it, would you be ok to just make us (or just me) a quick crash course on all these parameters ? What they mean and what are the choices you chose and why ? It can still be quick :D And while I think of it, even better... put that in the doc (comment) !